### PR TITLE
feat: [CHK-3284] add user-stats collection

### PR DESCRIPTION
**Pull request by scaminati-bv**
### List of changes
- add user-stats collection

### Motivation and context
The goal of this PR is to add **user-stats** collection to manage last payment method used by user

### Type of changes

- [x] Add new resources
- [ ] Update configuration to existing resources
- [ ] Remove existing resources

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [x] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [x] No

### Other information

---

### CODE REVIEW

File: src/domains/ecommerce-common/03_cosmosdb.tf [hunk 1]
```diff
@@ -143,6 +143,15 @@ locals {
...
       shard_key = "_id"
     },
+    {
+      name = "user-stats"
+      indexes = [{
+        keys   = ["_id"]
+        unique = true
+        }
+      ]
+      shard_key = "_id"
```
Comment (infantesimone) [inline src/domains/ecommerce-common/03_cosmosdb.tf:153]: what is this id? does it reflect these characteristics?
https://www.mongodb.com/docs/manual/core/sharding-choose-a-shard-key/

Comment (pietro-tota) [reply src/domains/ecommerce-common/03_cosmosdb.tf:153]: this id is the userId as returned by PDV, that is an UUID associated to the user fiscal code tokenization

**Review by pietro-tota — DISMISSED** [review]

File: src/domains/ecommerce-common/03_cosmosdb.tf [hunk 2]
```diff
@@ -143,6 +143,15 @@ locals {
       ]
       shard_key = "_id"
     },
+    {
+      name = "user-stats"
```
Comment (infantesimone) [inline src/domains/ecommerce-common/03_cosmosdb.tf:147]: retention on this data? To delete records of inactive users? has it been considered?

Comment (pietro-tota) [reply src/domains/ecommerce-common/03_cosmosdb.tf:151]: as per latest clarifications set TTL to 1 year with https://github.com/pagopa/pagopa-infra/pull/2453/commits/90b37decb1b2469cafe084b7870ea83594f441aa

**Comment by infantesimone** [comment]
could it be useful to have some index? maybe no :-)

**Review by pietro-tota — APPROVED** [review]

**Review by infantesimone — APPROVED** [review]

**Pull request merged**